app.ts

```diff
@@ -0,0 +1,238 @@
+				const message = `An error has occured: ${response.status}`;
+				throw new Error(message);
+			} else {
+				return await response.json();
+				console.log(response);
+			}
+		}
+		// trigger async function
+		// log response or catch error of fetch promise
+		getColumnNamesCall().then(data => console.log(data)).catch(reason => console.log(reason.message))
+		async function getRecordsCall(fromID: number, toID: number): Promise<string[][]> {
+			const response = await fetch(`http://localhost:2050/records?from=${(fromID)}&to=${(toID)}`);
+			// let promise = new Promise((res, rej) => {
+			// 	setTimeout(() => res("Now it's done!"), 50)
+			// });
+			if (!response.ok) {
+				const message = `An error has occured: ${response.status}`;
+				throw new Error(message);
+			} else {
+				return await response.json();
+				console.log(response);
+			}
+		}
+		// function requestcolumns<Request>(
+		//     method: 'GET',
+		//     url: 'http://localhost:2050/columns',
+		//     content?: Request,
+		//     callback?: (response: Response) => void,
+		//     errorCallback?: (err: any) => void) {
+		// const request = new XMLHttpRequest();
+		// request.open(method, url, true);
+		// request.onload = function () {
+		//     if (this.status >= 200 && this.status < 400) {
+		//         // Success!
+		//         const data = JSON.parse(this.response) as Response;
+		//         callback && callback(data);
+		//     } else {
+		//         // We reached our target server, but it returned an error
+
+		//         console.log("Error 404");
+		//     }
+		// };
+		// }
+		//region Data Loading methods
+		async function LoadRecordsData(fromID: number, toID: number): Promise<number[]> {
+			try {
+				const recordsvalue = await getRecordsCall(fromID, toID)
+				let DisplayContent = '';
+				for (const record of recordsvalue) {
+					DisplayContent += `<tr id="table-row-${record[0]}">`;
+					for (const column of record) {
+						DisplayContent += `<td align="center">${column}</td>`;
+					}
+					DisplayContent += '</tr>';
+					$("#wrapper-table-content-body").empty();
+					$("#wrapper-table-content-body").append(DisplayContent);
+				}
+			} catch (error) {
+				console.log("Error <No data>" + error)
+			}
+			return [fromID, toID];
+		}
+		function RecordsFromCursor(cursor: number[]): Promise<number[]> {
+			cursor = cursor.sort((a, b) => { return a - b });
+			return LoadRecordsData(cursor[0], cursor[1]);
+			// throw new Error("Error");
+		}
+		window.onresize = () => {
+			try {
+				const nextToId = calculateToId(previous[0]);
+				clearTimeout(Load);
+				Load = setTimeout(async () => {
+					const recordCount = await getRecordCountCall();
+					if (nextToId >= recordCount - 1) {
+						const fromId = recordCount - 1 - (calculateToId(previous[0]) - previous[0]);
+						const toId = recordCount - 1;
+						previous = await LoadRecordsData(fromId, toId);
+					} else {
+						previous = await LoadRecordsData(previous[0], nextToId)
+					}
+				}, 10);
```

Comment:
10 is quite quick... You should change it to 250. Anything around the human response time. Or at least, what is considered to be the average.

Reply:
the average is 250

Reply:
An average timeout is 250